src/genetic_low_memory.c

```diff
@@ -0,0 +1,459 @@
+  (*nextGenerationSize)++;
+  basePositionning++;
+}
+
+/**
+ *
+ * We create the children based off of a uniform crossover with two parents and
+ * apply the genetic operators
+ *
+ * @param firstParent  the first selected parent solution
+ * @param secondParent  the second selected parent solution
+ * @param nextGeneration the array containing all of the created chilren
+ * @param nextGenerationSize the amount of values in the array of created
+ * children
+ */
+static void createChildren(char* firstParent, char* secondParent,
+                           uint16_t* nextGeneration,
+                           unsigned int* nextGenerationSize) {
+
+  const unsigned int arraySize = UINT16_DIGIT_COUNT * dimensions;
+  const unsigned int encodedChildStringSize = arraySize + 1;
+
+  char firstChildString[encodedChildStringSize];
+  char secondChildString[encodedChildStringSize];
+
+  firstChildString[arraySize] = '\0';
+  secondChildString[arraySize] = '\0';
+
+  for (unsigned int i = 0; i < (UINT16_DIGIT_COUNT * dimensions); i++) {
+
+    if (linear_congruential_random_generator() <= 0.5) {
+
+      firstChildString[i] = firstParent[i];
+      secondChildString[i] = secondParent[i];
+    } else {
+      firstChildString[i] = secondParent[i];
+      secondChildString[i] = firstParent[i];
+    }
+  }
+
+  // We apply the mutation operator to both children
+  mutate(firstChildString, arraySize);
+  mutate(secondChildString, arraySize);
+
+  // We decode both children and add them to the next generation
+  decodeAndAddChild(nextGeneration, nextGenerationSize, firstChildString);
+  decodeAndAddChild(nextGeneration, nextGenerationSize, secondChildString);
+}
+
+/**
+ * We encode a solution as a string , the given string will always be the same
+ * length
+ *
+ * @param combinedValue the created string containing the encoded parent
+ * @param parent an array containing the parameters of the parent
+ */
+static void encode(char* combinedValue, uint16_t* parent) {
+
+  unsigned int arrIndex = 0;
+  unsigned int encodedParameterSize = UINT16_DIGIT_COUNT + 1;
+
+  for (unsigned int i = 0; i < dimensions; i++) {
+
+    char stringValue[encodedParameterSize];
+    char otherString[encodedParameterSize];
+
+    stringValue[UINT16_DIGIT_COUNT] = '\0';
+    otherString[UINT16_DIGIT_COUNT] = '\0';
+
+    unsigned int otherStringLength = 0;
+    for (int j = (UINT16_DIGIT_COUNT - 1); j >= 0; j--) {
+
+      if (parent[i] < pow(10, j)) {
+        otherString[otherStringLength] = '0';
+        otherStringLength++;
+      } else {
+        break;
+      }
+    }
+
+    sprintf(stringValue, "%hu", parent[i]);
+
+    if (otherStringLength != 0) {
+      for (unsigned int j = 0; j < otherStringLength; j++) {
+        combinedValue[j + arrIndex] = otherString[j];
+      }
+    }
+
+    for (unsigned int j = otherStringLength; j < UINT16_DIGIT_COUNT; j++) {
+
+      combinedValue[j + arrIndex] = stringValue[j - otherStringLength];
+    }
+    arrIndex += UINT16_DIGIT_COUNT;
```

Comment:
arrIndex is equal to (i * UINT16_DIGIT_COUNT) which both take 1 cycle to compute. Please replace this variable with a const that is equal to the given equation.

Reply:
(It's safer and easier to read)